### PR TITLE
Add testonly to libraries that link XCTest

### PR DESCRIPTION
**Pull request by dierksen**
Tested locally to ensure that `testonly = True` is added only to the appropriate targets.

### CODE REVIEW

**Review by thiagohmcruz — APPROVED** [review]
Aren't there any tests for this? If not maybe we can add one?

**Review by luispadron — APPROVED** [review]
Nice we're doing this in our rules currently but this makes more sense.

**Review by justinseanmartin — APPROVED** [review]

File: lib/cocoapods/bazel/target.rb [hunk 1]
```diff
@@ -374,8 +374,10 @@ def to_rule_kwargs
         kwargs[:runtime_deps] = []
         kwargs[:sdk_dylibs] = file_accessors.flat_map { |fa| fa.spec_consumer.libraries }.sort.uniq
         kwargs[:sdk_frameworks] = file_accessors.flat_map { |fa| fa.spec_consumer.frameworks }.sort.uniq
+        kwargs[:testonly] = true if kwargs[:sdk_frameworks].include? 'XCTest'
```
Comment (jerrymarino) [inline lib/cocoapods/bazel/target.rb:377]: @dierksen is this for the explict PCM stuff? I think the issue there is we need to add framework PCMS based on `sdk_frameworks`  https://github.com/bazel-ios/rules_ios/pull/562#discussion_r1022122823

There are other reasons we want to do this - but it also prevents having to do something like this

Comment (dierksen) [reply lib/cocoapods/bazel/target.rb:377]: no, this is to account for changes in https://github.com/bazelbuild/rules_swift/pull/868 that are preventing us from bumping `rules_ios`.

Comment (jerrymarino) [reply lib/cocoapods/bazel/target.rb:377]: Well it looks like that PR is going to force you to use `test_only` all of the place 🤔 

Maybe the same issue applies: e.g. adding the paths more tactfully. I know one sticking point here is wanting to use `XCTest` in other contexts

Comment (justinseanmartin) [reply lib/cocoapods/bazel/target.rb:377]: `wanting to use XCTest in other contexts` inside of binaries that ship with the app you mean? I'm not sure I see the harm in the change here, but I might not be understanding something.

Comment (jszumski) [reply lib/cocoapods/bazel/target.rb:377]: FWIW we had to split a few modules into two halves: one for mocks/fakes that were used in test apps and one for assertion utilities that were used in unit tests and thus got `testonly` to import XCTest

**Comment by jerrymarino** [comment]
@justinseanmartin I wonder if can add thse flags differently based on what Maxwell added to rules_swift to get it working with pods

What do you think if we added back the old behavior into rules_ios conditionally - or adding more intricate logic to detect adding it? I'm still not 100% sure the original motivation to gut the flags.

File: lib/cocoapods/bazel/target.rb (hunk 1 above)
Comment (jerrymarino) [reply lib/cocoapods/bazel/target.rb:377]: >  I'm not sure I see the harm in the change here,

➕  I'm curious to know as well. Also how this overlaps with cocoapods/Xcode defaults. One thing I thought of was when I end up using `XCTest` framework elsewhere than a test - or you have tests that don't use `XCTest` ..

**Comment by jerrymarino** [comment]
ref: https://github.com/bazelbuild/rules_swift/pull/868#issuecomment-1320412612

File: lib/cocoapods/bazel/target.rb (hunk 1 above)
Comment (luispadron) [reply lib/cocoapods/bazel/target.rb:377]: The rules_swift change made it so any target that wishes to link XCTest must be marked testonly. 

This change correctly applies the testonly attribute to the BUILD files in order to work with the new rules_swift changes.

**Comment by luispadron** [comment]
We'll need to update `rules_ios` since this now passes `testonly` for unit test targets which causes duplicate args because `rules_ios` always passes `testonly = True` for all test targets as well.

```

00:05:44.389 	File "/private/var/tmp/_bazel_build/5f8e42350ee950cd952d5c32bb6b1f7e/external/build_bazel_rules_ios/rules/test.bzl", line 61, column 28, in _ios_test
00:05:44.389 		library = apple_library(name = name, namespace_is_module_name = False, platforms = {"ios": ios_test_kwargs.get("minimum_os_version")}, testonly = True, **kwargs)
00:05:44.389 	File "/Users/build/.jenkins/workspace/PR+cas+BCVAFG+PR-Bazel-Optional/ios-builder/s/c/Tools/Rules/library.bzl", line 14, column 5

```

File: lib/cocoapods/bazel/target.rb (hunk 1 above)
Comment (jerrymarino) [reply lib/cocoapods/bazel/target.rb:377]: Cool - makes sense to me: re - making it work with latest rules_swift.

I was thinking if it'd be worth to improve on that - e.g. if there is another way: e.g. auto-detecting adding the search path when you use `XCTest` VS adding other hinges no `testonly`.. 

I personally wouldn't bind the 2 together. Purity aside today when a developer adds `XCTest` to some `apple_framework` in rules_ios today it kind of just works - but now we'll expect them to fix these new errors by having to add `testonly`.. 

That said, you can probably preserve the old behavior by just injecting `testonly` in rules_ios VS doing analysis phase detection of it.

**Comment by luispadron** [comment]
This should fix the above issue: https://github.com/bazel-ios/rules_ios/pull/609

**Pull request merged**